### PR TITLE
[SPARK-40445][PS] Refactor `Resampler` for consistency and simplicity

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Refactor `Resampler` to make it consistent with `GroupBy`

### Why are the changes needed?
for consistency (more like `GroupBy`) and simplicity

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
existing UT

### CODE REVIEW

**Comment by zhengruifeng** [comment]
cc @HyukjinKwon

File: python/pyspark/pandas/resample.py [hunk 1]
```diff
@@ -481,20 +489,5 @@ def __getattr__(self, item: str) -> Any:
...
-    def mean(self) -> Series:
-        return first_series(self._downsample("mean"))
-
-    def std(self) -> Series:
-        return first_series(self._downsample("std"))
-
-    def var(self) -> Series:
-        return first_series(self._downsample("var"))
+    def _cleanup_and_return(self, psdf: DataFrame) -> Series:
+        return first_series(psdf).rename().rename(self._psser.name)
```
Comment (itholic) [inline python/pyspark/pandas/resample.py:493]: qq: Why we do rename twice here?

**Review by itholic — APPROVED** [review]
Looks good otherwise

File: python/pyspark/pandas/resample.py (hunk 1 above)
Comment (zhengruifeng) [reply python/pyspark/pandas/resample.py:493]: good question: I just copy it from `groupby`, let me check whether the first `rename` can be removed

File: python/pyspark/pandas/groupby.py [hunk 2]
```diff
@@ -3762,7 +3762,7 @@ def _apply_series_op(
             return psser.copy()
 
     def _cleanup_and_return(self, psdf: DataFrame) -> Series:
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/groupby.py:3764]: Should probably rename this to something else ... e.g., `_handle_output`

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise

File: python/pyspark/pandas/groupby.py (hunk 2 above)
Comment (zhengruifeng) [reply python/pyspark/pandas/groupby.py:3764]: nice, let me update it

**Comment by zhengruifeng** [comment]
Merged into master, thank you @itholic @HyukjinKwon

**Pull request closed without merging**